include/xgboost/learner.h

```diff
@@ -191,5 +178,57 @@ class Learner : public rabit::Serializable {
   GenericParameter generic_param_;
 };
 
+
+/*! \brief training parameter for regression */
+struct LearnerModelParam : public dmlc::Parameter<LearnerModelParam> {
+  /* \brief global bias */
+  bst_float base_score;
+  /* \brief number of features  */
+  unsigned num_feature;
+  /* \brief number of classes, if it is multi-class classification  */
+  int num_class;
+  /*! \brief Model contain additional properties */
+  int contain_extra_attrs;
+  /*! \brief Model contain eval metrics */
+  int contain_eval_metrics;
+  /*! \brief reserved field */
+  int reserved[29];
```

Comment:
Is there any reason to make it this size?
I think this deserves a comment and maybe a static_assert checking that the size of a structure is not too big

Reply:
@trams No idea.  It's legacy code for binary IO.  This PR is setting up everything to remove it.

Reply:
@trams 

> I do not want to nitpick. I am just a bit confused. It seems this include is still here. Do I miss some other pull request or updates?

Sorry about that.  It's not entering the "clean up" phase yet.  I kept finding new problems in the old binary IO and parameter logics.  Excited to get them fixed.